### PR TITLE
Patch for issue #101

### LINKED ISSUE #101: Small security issue - Caddy jwt-secret value - If default is not used the first jwt-secret is used
---**This issue could be a breaking change depending on the caddyfile and exact setup**---

My use-case is to have two subdomains providing different content and use two different secret keys.

I tested this minimal caddyfile on Windows 7 x64:

```
http://sub1.localhost:2015 {
	root "C:\sites\1"

	jwt {
		path /members
		redirect /login
		secret "C:\sites\key1.txt"
	}

	login {
        	success_url /members/phpinfo.php
        	htpasswd file=users1.txt
		jwt-secret aaaaaaa
		cookie-domain sub1.localhost
	}
}

http://sub2.localhost:2015 {
	root "C:\sites\2"

	jwt {
		path /members
		redirect /login
		secret "C:\sites\key2.txt"
	}

	login {
        success_url /members/phpinfo.php
        htpasswd file=users2.txt
	jwt-secret bbbbbbb
	cookie-domain sub2.localhost
    }
}
```

I checked the produced token there: https://jwt.io/

I noticed the first `jwt-secret` ('aaaaaaa') is used for both subdomains.

I'm investigating this issue and I'll have a PR in the next few hours.

### PR DESCRIPTION
**Pull request by magikstm**
Fixes #101.

This PR may need tests, more user tests and a review to confirm it fits expected behavior.

@smancke please review for next release if possible.

### CODE REVIEW

**Comment by smancke** [comment]
Hey @magikstm 
I think, this PR has two problems: 
1. If a secret is set by environment variable, from outside, it is not used.
2. If the default secret is used, it does not get populated to caddy.jwt, because in the case of the default, the env never gets set.
(It would be good to have this requirements tested as well.)

I think, a solution would be, to check, if there is an OS variable set from outside, before the configuration handling is done. e.g. set a global variable in an init() function of the file.
Than the code in the configuration can distinguish between a variable set from outside and a variable set from a previous configuration.

File: README.md [hunk 1]
```diff
@@ -44,38 +44,38 @@ For questions and support please use the [Gitter chat room](https://gitter.im/ta
...
+| -facebook                   | value       |              | X     | OAuth config in the form: client_id=..,client_secret=..,scope=email..[redirect_uri=..]                                                                      |
+| -host                       | string      | "localhost"  | -     | Host to listen on                                                                                                                                           |
+| -htpasswd                   | value       |              | X     | Htpasswd login backend opts: file=/path/to/pwdfile                                                                                                          |
+| -jwt-expiry                 | go duration | 24h          | X     | Expiry duration for the JWT token, e.g. 2h or 3h30m                                                                                                         |
+| -jwt-secret                 | string      | "random key" | X     | Secret used to sign the JWT token (if you don't use default, set related secret in http.jwt) (you can't use default and a custom one in the same caddyfile) |
```
Comment (smancke) [inline README.md]: Better keep this string short here and mark it with a footnote where the different possibilities are explained in more detail. And be aware in the docu, that 50% of the loginsrv users use it standalone and should not care about caddy. They should not get confused by hints to http.jwt, which they properly do not know at all.

Comment (magikstm) [reply README.md]: Thanks for the note. I'll adjust doc differently.

**Comment by magikstm** [comment]
Thanks the review. I'll edit this PR.

**Comment by magikstm** [comment]
@smancke please let me know if last commit looks good. If it is, I'll add documentation and try to add tests.

I could add these notes in README.md:

> **JWT_TOKEN handling**
> - If a `JWT_TOKEN` environment variable is set it is used by http.login
> - If `-jwt-secret`  default "random key" is used by http.login it is set as `JWT_TOKEN` environment variable
> 
> (*) if you use Caddy and don't use default, set related `secret` in http.jwt. You can't use default and a custom one in the same caddyfile.

**Comment by smancke** [comment]
@magikstm I think, it is still not correct ..  let's see if I get the time to work on this ..

**Comment by magikstm** [comment]
I'm closing this PR. PR #113 is a better solution, includes tests as well as related documentation changes.

**Pull request closed without merging**